cli/standalone/mod.rs

```diff
@@ -150,12 +149,22 @@ impl ModuleLoader for EmbeddedModuleLoader {
     is_dynamic: bool,
     _requested_module_type: RequestedModuleType,
   ) -> deno_core::ModuleLoadResponse {
-    let is_data_uri = get_source_from_data_url(original_specifier).ok();
-    if let Some((source, _)) = is_data_uri {
+    if original_specifier.scheme() == "data" {
+      let data_url_text =
+        match deno_graph::source::RawDataUrl::parse(original_specifier)
```

Comment:
I'm a bit confused here - if we use `RawDataUrl` then why do we need to check for specifier scheme? Maybe this struct should be renamed after all?

Reply:
This function is the general loader function. I think it's better to surface data uri errors if they happen rather than it heading below as it was doing before (and I'm not sure what kind of behaviour would occur). Maybe I missed something in the code here though?

Reply:
That makes sense and is actually better 👍

Reply:
I'm going to try to add a test for this then merge this pr.

Reply:
Before:

```
error: Uncaught (in promise) TypeError: Module not found: data:application/
await import('data:application/');
^
    at async file:///C:/Users/david/AppData/Local/Temp/deno-cli-testSjmgsB/bad_data_uri.ts:1:1
```

After:

```
error: Uncaught (in promise) TypeError: Unable to decode data url.
await import('data:application/');
^
    at async file:///C:/Users/david/AppData/Local/Temp/deno-cli-test5fjBTc/bad_data_uri.ts:1:1
```

Added a test.